client/src/app/mainwindow.cpp

```diff
@@ -723,6 +724,15 @@ int MainWindow::resourceIndexFor(const QString &id) const
     return -1;
 }
 
+void MainWindow::raiseWidget(QWidget *widget)
```

Comment:
The method name is confusing. This does more than raising that widget, it also raises the mainwindow.

raiseMainWindowAndDialog() ?